### PR TITLE
Recognize experimental C++ interoperability flags

### PR DESCRIPTION
**Pull request by artemcm**

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci please test

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 1]
```diff
@@ -3017,6 +3017,29 @@ final class SwiftDriverTests: XCTestCase {
     }
   }
 
+  func testCXXInteropOptions() throws {
+    do {
+      var driver = try Driver(args: ["swiftc", "-enable-experimental-cxx-interop", "foo.swift"])
+      let plannedJobs = try driver.planBuild()
```
Comment (owenv) [inline Tests/SwiftDriverTests/SwiftDriverTests.swift:3023]: This and line 3033 probably just need a trailing `removingAutolinkExtractJobs` to work on Linux

**Comment by artemcm** [comment]
@swift-ci please test

File: Tests/SwiftDriverTests/SwiftDriverTests.swift (hunk 1 above)
Comment (artemcm) [reply Tests/SwiftDriverTests/SwiftDriverTests.swift]: Good catch, thanks!

**Review by DougGregor — APPROVED** [review]

**Review by owenv — APPROVED** [review]

**Comment by artemcm** [comment]
@swift-ci please test

**Comment by artemcm** [comment]
@swift-ci please test

**Comment by artemcm** [comment]
@swift-ci please test

**Pull request merged**